drivers/sensor/lis2dw12/lis2dw12_trigger.c

```diff
@@ -281,5 +284,13 @@ int lis2dw12_init_interrupt(struct device *dev)
 		return -EIO;
 	}
 
-	return gpio_pin_enable_callback(lis2dw12->gpio, cfg->int_gpio_pin);
+	/* set interrupt polarity */
+	int_polarity = ((cfg->int_gpio_flags & BIT(0)) == GPIO_ACTIVE_HIGH) ?
+				LIS2DW12_ACTIVE_HIGH : LIS2DW12_ACTIVE_LOW;
+	if (lis2dw12_pin_polarity_set(lis2dw12->ctx, int_polarity)) {
```

Comment:
We should use the flag that is not `0` to perform the test, as in
```
	int_polarity = ((cfg->int_gpio_flags & GPIO_ACTIVE_LOW) != 0) ?
￼				LIS2DW12_ACTIVE_LOW : LIS2DW12_ACTIVE_HIGH;
```

Reply:
Well, I changed it recently.
Before was:

    int_polarity = ((cfg->int_gpio_flags & GPIO_ACTIVE_LOW) == GPIO_ACTIVE_LOW) ? 
                            LIS2DW12_ACTIVE_LOW : LIS2DW12_ACTIVE_HIGH;

If you agree, I would do it like this again.

Reply:
The way proposed as standard in https://github.com/zephyrproject-rtos/zephyr/issues/20017#issuecomment-549315497 is, where the hardware allows control of IRQ level, to leave it at the device power-on default and use the devicetree flags to adjust the sense of the detection.  And update the yaml to document the required sense.

Then you can get rid of the polarity set API.

Reply:
> The way proposed as standard in [#20017 (comment)](https://github.com/zephyrproject-rtos/zephyr/issues/20017#issuecomment-549315497) is, where the hardware allows control of IRQ level, to leave it at the device power-on default and use the devicetree flags to adjust the sense of the detection. And update the yaml to document the required sense.
> 
> Then you can get rid of the polarity set API.

@pabigot 
I'm not able to spot the polarity standard in the #20017 comment.
Anyway, I read your point here but I'm not sure I got it properly. You said that we should adjust the sense of detection using dts flags (provided that the h/w allows it), but then you said that we can get rid of the polarity set API. Can you clarify this point? Not sure I got it.

Reply:
Yeah, it's not explicit: it's implicit in the referenced hts221 driver update.

See https://github.com/zephyrproject-rtos/zephyr/blob/fc4c349dcd5eee7ab77cf00e79787a6ce4ebd37a/dts/bindings/sensor/st%2Chts221.yaml#L20

If the signal polarity is left at default you don't need to set it in the driver.

Reply:
> Yeah, it's not explicit: it's implicit in the referenced hts221 driver update.
> 
> See
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/fc4c349dcd5eee7ab77cf00e79787a6ce4ebd37a/dts/bindings/sensor/st%2Chts221.yaml#L20
> 
> If the signal polarity is left at default you don't need to set it in the driver.

Ok, got it.
Stupid question: is it possible that a given board with that sensor, for ANY reason, invert the polarity of the drdy? In that case you should configure in the dts ACTIVE_LOW. I know that it shouldn't be, but I guess that setting the polarity in the driver according to what defined in the dts is a good approach.

Reply:
It is possible that the signal would be inverted on a board, but that would be handled by correcting the active level in devicetree.  I haven't come up with any reason to ever change the polarity configuration in the sensor itself from its default.  It should *not* be configured to match the polarity specified in devicetree, because we can handle the two options by respecting the gpio configured active level.

Reply:
> It is possible that the signal would be inverted on a board, but that would be handled by correcting the active level in devicetree. I haven't come up with any reason to ever change the polarity configuration in the sensor itself from its default. It should _not_ be configured to match the polarity specified in devicetree, because we can handle the two options by respecting the gpio configured active level.

Clear. The gpio_pin_configure() already set the polarity in the way the dts dictate. And the dts is just a description of the h/w. I will remove the lis2dw12_pin_polarity_set() API.
Thx!